pkg/server/file.go

```diff
@@ -152,21 +151,28 @@ func (t *TranslateServiceServer) UploadTranslationFile(
 				continue
 			}
 
-			err = t.repo.SaveMessages(ctx, params.serviceID, &all[i])
-			switch {
-			default:
-				// noop
-			case errors.Is(err, repo.ErrNotFound):
-				return nil, status.Errorf(codes.NotFound, "service not found")
-			case err != nil:
-				return nil, status.Errorf(codes.Internal, "")
+			empty, err, done = t.SaveMsgsAndHandleErrors(ctx, params, messages)
+			if done {
+				return empty, err
 			}
 		}
 	}
 
 	return &emptypb.Empty{}, nil
 }
 
+func (t *TranslateServiceServer) SaveMsgsAndHandleErrors(ctx context.Context, params *uploadParams, messages *model.Messages) (*emptypb.Empty, error, bool) {
+	switch err := t.repo.SaveMessages(ctx, params.serviceID, messages); {
+	default:
+		// noop
+	case errors.Is(err, repo.ErrNotFound):
+		return nil, status.Errorf(codes.NotFound, "service not found"), true
+	case err != nil:
+		return nil, status.Errorf(codes.Internal, ""), true
+	}
+	return nil, nil, false
+}
+
```

Comment:
I think this function is not needed, as it does not benefit code readability, at least in its current state.
There are many things wrong with that:
1. **Function Naming**: Functions must do one thing, e.g. **Single Responsibility Principle**, and when you use `and` in the function name, it usually implies that it breaks that principle
2. For return values, only `error` would be enough -> `func(...) error {}`
3. No reason to ignore `default` case, and then executing `default` statement outside `switch` statement